kafka-client/src/main/java/dev/responsive/kafka/internal/clients/OriginEventRecorder.java

```diff
@@ -0,0 +1,210 @@
+    private final String licenseServer;
+    private final String apiKey;
+    private final String appId;
+    private final String env;
+
+    private Reporter(ResponsiveConfig config, final LicenseInfo license) {
+      if (!license.type().equals(UsageBasedV1.TYPE_NAME)) {
+        throw new IllegalArgumentException(
+            "Invalid license type for usage reporting: " + license.type());
+      }
+
+      this.licenseServer = config.getString(ResponsiveConfig.RESPONSIVE_LICENSE_SERVER_CONFIG);
+      this.apiKey = ((UsageBasedV1) license).key();
+      this.appId = (String) config.originals().get(StreamsConfig.APPLICATION_ID_CONFIG);
+      this.env = config.getString(ResponsiveConfig.RESPONSIVE_ENV_CONFIG);
+
+      client = HttpClient.newHttpClient();
+    }
+
+    @Override
+    public void accept(final Map<TopicPartition, Long> counts, final String threadId) {
+      LOG.debug("Reporting origin events for {} on {}", counts, threadId);
+
+      final var body = OriginEventsReportRequestV1.builder()
+          .setTimestamp(System.currentTimeMillis())
+          .setTransactionId(UUID.randomUUID().toString())
+          .setEventCount(counts.values().stream().reduce(Long::sum).orElse(0L))
+          .setEnv(env)
+          .setApplicationId(appId)
+          .setThreadId(threadId)
+          .build();
+
+      try {
+        final HttpRequest request = HttpRequest.newBuilder()
+            .uri(new URI(licenseServer + "/usage"))
+            .header("Content-Type", "application/json")
+            .header("Authorization", "Bearer " + apiKey)
+            .POST(HttpRequest.BodyPublishers.ofString(mapper.writeValueAsString(body)))
+            .build();
+
+        client.sendAsync(request, HttpResponse.BodyHandlers.ofString())
+            .thenAccept(response -> {
+              final int status = response.statusCode();
+              final String responseBody = response.body();
+              if (status >= 400 && status < 500) {
+                LOG.error(
+                    "Failed to report usage due to {} (error code {}). "
+                        + "Please make sure your license is correctly configured to "
+                        + "avoid violation of the license agreement terms.",
+                    responseBody,
+                    status
+                );
+              } else if (status >= 500 && status < 600) {
+                LOG.warn("Failed to report usage due to {} (code {}).", status, responseBody);
+              } else {
+                LOG.debug("Successfully reported usage (status: {}): {}", status, responseBody);
+              }
+            })
+            .exceptionally(e -> {
+              LOG.error("Failed to report usage! Please make sure "
+                  + "you have correctly configured your Responsive License", e);
+              return null;
+            });
+      } catch (final JsonProcessingException | URISyntaxException e) {
+        // this should never happen
```

Comment:
we should probably just log in this case right? We don't want to fail the app if there's some bug here

Reply:
I split the two - if there's a URI syntax error then it's a problem with the configuration which users should change.